### PR TITLE
Feat: Documentation for Monolake

### PR DESCRIPTION
**Pull request by har23k**
Documentation for Monolake: https://github.com/cloudwego/monolake

### CODE REVIEW

File: content/en/docs/monolake/Overview/_index.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+| [certain-map](https://github.com/ihciah/certain-map) | A foundational crate that provides a typed map data structure, ensuring the existence of specific items at compile-time, useful for managing data dependencies between services |
+| [monoio-thrift](https://github.com/monoio-rs/monoio-thrift) | Monoio native, io_uring compatible thrift implementation |
+| [monoio-http](https://github.com/monoio-rs/monoio-http) | Monoio native, io_uring compatible HTTP/1.1 and HTTP/2 implementation |
+| [monoio-nativetls](https://github.com/monoio-rs/monoio-tls) | The native-tls implementation compatible with monoio |
+| [monoio-rustls](https://github.com/monoio-rs/monoio-tls) | The rustls implementation compatible with monoio |
+
+## Blogs
+
+- [Monolake: How ByteDance Developed Its Own Rust Proxy to Save Hundreds of Thousands of CPU Cores](TODO)
+- [Monolake open source summit](TODO)
```
Comment (GuangmingLuo) [inline content/en/docs/monolake/Overview/_index.md:71]: Please remove this for now

File: content/en/docs/monolake/Tutorial/service.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+---
+title: "Creating Service and Factory"
+linkTitle: "Creating Service and Factory"
+weight: 2
+---
+
+## Prerequisites
+For a better
```
Comment (GuangmingLuo) [inline content/en/docs/monolake/Tutorial/service.md:8]: This is imcomplete

File: content/en/docs/monolake/Config Guide/_index.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
+---
+title: "Config Guide"
+linkTitle: "Config Guide"
+weight: 5
+date: 2023-07-3
```
Comment (GuangmingLuo) [inline content/en/docs/monolake/Config Guide/_index.md:5]: please update the date

File: content/en/docs/monolake/FAQ/_index.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+title: "FAQ"
+linkTitle: "FAQ"
+date: 2023-07-03
```
Comment (GuangmingLuo) [inline content/en/docs/monolake/FAQ/_index.md:4]: update

File: content/en/docs/monolake/Architecture/context.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+## The Problem `certain_map` Solves
+
+When building modular services, it's common to have indirect data dependencies between components. For example, a downstream service may require information that was originally provided in an upstream request, but the intermediate services don't directly use that data. Traditionally, this would involve passing all potentially relevant data through the request/response types, which can quickly become unwieldy and error-prone.
+
+Alternatively, you might use a `HashMap` to manage the context data, but this approach has a significant drawback: you cannot ensure at compile-time that the required key-value pairs have been set when the data is read. This can lead to unnecessary error handling branches or even panics in your program.
+
+## How `certain_map` Helps
+
+The `certain_map` crate solves this problem by providing a typed map that ensures the existence of specific items at compile-time. When you define a `Context` struct using `certain_map`, the compiler will enforce that certain fields are present, preventing runtime errors and simplifying the implementation of your services.
```
Comment (ihciah) [inline content/en/docs/monolake/Architecture/context.md:19]: `a typed map`, to be more precise it is not a typed map but just for users it's like that. We can describe it as `a typed-map-like struct`

**Review by ihciah — APPROVED** [review]

**Review by GuangmingLuo — APPROVED** [review]

**Pull request merged**